src/pydata_sphinx_theme/assets/styles/extensions/_sphinx_design.scss

```diff
@@ -50,32 +41,63 @@ $sd-semantic-color-names: (
   * already define for our own needs.
   */
 $extra-semantic-colors: (
+  // TODO need to better adjust these colours (dark and muted)
   "dark": map-get($pst-semantic-colors, "text-base"),
   "muted": map-get($pst-semantic-colors, "text-muted"),
-  "light": rgb(201, 201, 201),
-  "black": rgb(0, 0, 0),
-  "white": rgb(255, 255, 255),
+  "light": $foundation-light-grey,
+  "black": $foundation-black,
+  "white": $foundation-white
 );
 
 $all-colors: map-merge($pst-semantic-colors, $extra-semantic-colors);
 
+@mixin create-sd-colors($value, $name) {
+  // define the pst variables, so that downstream user overrides will work
+  --pst-color-#{$name}: #{$value};
+  @debug "created color 🌱";
+  // we are now using a11y-combination to calculate the text color - this is based
+  // on the WCAG color contrast guidelines
+  --pst-color-#{$name}-text: #{a11y-combination($value)};
```

Comment:
@choldgraf I made a drastic change on how to assign/define text colour (to meet WCAG contrast criteria), but since the method here was an adaptation of what sphinx design uses I am thinking of adding this new approach on sphinx design too